kernels/include/gnuradio/kernels/cpu/kernel.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+#include <cstddef>
+
+
+namespace gr {
+namespace kernels {
+
+enum DeviceType { CPU, GPU };
+enum DataType {
```

Comment:
Yes, we need a common way to map blocks/ports to kernels so we should use the same set of enums. To prevent kernels depending on gnuradio runtime/blocks, I'm thinking the enums should live in the gr::kernel namespace.